elfi/model/extensions.py

```diff
@@ -16,6 +23,7 @@ def __init__(self, name=None):
         """
         self._name = name or self.__class__.__name__
 
+    @classmethod
```

Comment:
The MA2 example won't work with the old version (`logpdf` calling `this.pdf`).

We can add a definition for non-static classes once we need it. Of course, even better if there's a solution that works for both.